pkg/chunked/internal/compression.go

```diff
@@ -167,13 +170,41 @@ func WriteZstdChunkedManifest(dest io.Writer, outMetadata map[string]string, off
 		return err
 	}
 
+	var compressedTarSplitBuffer bytes.Buffer
+	compressTarSplitWriter, err := ZstdWriterWithLevel(&compressedTarSplitBuffer, level)
+	if err != nil {
+		return err
+	}
+	if _, err := compressTarSplitWriter.Write(tarSplitData); err != nil {
+		compressTarSplitWriter.Close()
+		return err
+	}
+	if err := compressTarSplitWriter.Close(); err != nil {
+		return err
+	}
+	compressedTarSplit := compressedTarSplitBuffer.Bytes()
+
+	tarSplitDigester := digest.Canonical.Digester()
+	tarSplitChecksum := tarSplitDigester.Hash()
+	if _, err := tarSplitChecksum.Write(compressedTarSplit); err != nil {
+		return err
+	}
+	outMetadata[TarSplitChecksumKey] = tarSplitDigester.Digest().String()
```

Comment:
This seems to amount to `digest.Canonical.FromBytes()`. If so, please use that.

(Alternatively, compute the digest on the fly while creating the data.)